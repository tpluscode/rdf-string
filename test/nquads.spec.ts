import $rdf from 'rdf-ext'
import namespace from '@rdfjs/namespace'
import { schema, xsd } from '@tpluscode/rdf-ns-builders'
import { expect } from 'chai'
import { nquads } from '../src/index.js'
import './matchers.js'

const ex = namespace('http://example.com/')

describe('nquads', () => {
  describe('interpolating JS types', () => {
    it('leaves strings intact', () => {
      // when
      const foo = 'foo'
      const str = nquads`<http://example.com/${foo}>`.toString()

      // then
      expect(str).to.equal('<http://example.com/foo>')
    })
  })

  describe('interpolating dataset', () => {
    it('writes all graphs', function () {
      // given
      const salary = $rdf.blankNode('salary')
      const dataset = $rdf.dataset()
        .add($rdf.quad(ex.John, schema.name, $rdf.literal('John'), ex.John))
        .add($rdf.quad(ex.Jane, schema.name, $rdf.literal('Jane'), ex.Jane))
        .add($rdf.quad(ex.Jane, schema.spouse, ex.Jane))
        .add($rdf.quad(ex.John, schema.baseSalary, salary, ex.John))
        .add($rdf.quad(salary, schema.value, $rdf.literal('20000', xsd.int), ex.John))

      // when
      const str = nquads`${dataset}`.toString()

      // then
      expect(str).to.matchSnapshot(this)
      expect(str).to.be.validNQuads()
    })

    it('does not break on empty dataset', () => {
      // given
      const dataset = $rdf.dataset()

      // when
      const str = nquads`${dataset}`.toString()

      // then
      expect(str).to.equal('')
    })
  })
})
