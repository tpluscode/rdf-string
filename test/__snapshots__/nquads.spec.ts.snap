exports["nquads interpolating dataset writes all graphs 1"] = "\n\"<http://example.com/John> <http://schema.org/name> \\\"John\\\" <http://example.com/John> .\n<http://example.com/John> <http://schema.org/baseSalary> _:salary <http://example.com/John> .\n_:salary <http://schema.org/value> \\\"20000\\\"^^<http://www.w3.org/2001/XMLSchema#int> <http://example.com/John> .\n<http://example.com/Jane> <http://schema.org/name> \\\"Jane\\\" <http://example.com/Jane> .\n<http://example.com/Jane> <http://schema.org/spouse> <http://example.com/Jane>  .\"\n";

exports["nquads interpolating dataset writes all graphs"] = "<http://example.com/John> <http://schema.org/name> \"John\" <http://example.com/John> .\n<http://example.com/John> <http://schema.org/baseSalary> _:salary <http://example.com/John> .\n_:salary <http://schema.org/value> \"20000\"^^<http://www.w3.org/2001/XMLSchema#int> <http://example.com/John> .\n<http://example.com/Jane> <http://schema.org/name> \"Jane\" <http://example.com/Jane> .\n<http://example.com/Jane> <http://schema.org/spouse> <http://example.com/Jane>  .";

